File: Backend/src/main/java/com/example/BTL_Mobile/dto/response/lesson/QuestionDetailDTO.java
package com.example.BTL_Mobile.dto.response.lesson;

import com.example.BTL_Mobile.dto.response.MediaDTO;
import com.example.BTL_Mobile.model.enums.EQuestionType;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class QuestionDetailDTO {

    private int id;

    private String content;

    private EQuestionType type;

    private MediaDTO media;

    private List<AnswerDTO> answers;

    private boolean repeatable;

}

File: Mobile/app/src/main/java/com/nhom2/elearnlanguage/data/dto/lesson/QuestionDetailDTO.kt
package com.nhom2.elearnlanguage.data.dto.lesson

import kotlinx.serialization.Serializable

@Serializable
data class QuestionDetailDTO(
    val id: Int,
    val type: String, // "ONE_SELECTION", "LISTEN_AND_ARRANGE_SENTENCE", "TRANSLATE_AND_ARRANGE_SENTENCE", "SPEAKING_ASSESSMENT"
    val content: String,
    val repeatable: Boolean = false,
    val media: MediaDTO? = null,
    val answers: List<AnswerDTO> = emptyList()
)
